hydrator-common/src/main/java/co/cask/hydrator/common/ReferenceBatchSource.java

```diff
@@ -38,5 +44,20 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
     // Verify that reference name meets dataset id constraints
     IdUtils.validateId(config.referenceName);
+    // create the external dataset this is done in the configure pipeline stage so that the external dataset with or
+    // without the schema is created when the pipeline is deployed itself.
+    Schema externalSchema = getSchema();
+    Map<String, String> dsProperties = Collections.emptyMap();
+    if (externalSchema != null) {
+      dsProperties = Collections.singletonMap(DatasetProperties.SCHEMA, externalSchema.toString());
+    }
+    pipelineConfigurer.createDataset(config.referenceName, Constants.EXTERNAL_DATASET_TYPE,
+                                     DatasetProperties.of(dsProperties));
   }
+
+  /**
+   * @return The schema of the source if known else null
+   */
+  @Nullable
+  public abstract Schema getSchema();
```

Comment:
This encourages plugins to put schema parsing logic in their class instead of in their plugin config. In general, we want the config object to expose a getSchema() method so that the string value of the schema is never used. I think it makes more sense to put this in the config class and not have it as abstract, returning null by default.